src/teleport.cpp

```diff
@@ -0,0 +1,93 @@
+#include "teleport.h"
+
+#include "character.h"
+#include "avatar.h"
+#include "creature.h"
+#include "player.h"
+#include "monster.h"
+#include "game.h"
+#include "map.h"
+#include "messages.h"
+#include "point.h"
+
+
+const efftype_id effect_teleglow( "teleglow" );
+
+bool teleport::teleport( Creature *c, int min_distance, int max_distance, bool safe,
```

Comment:
i would prefer a variable name that's more than one letter, like `critter`. also, is there a particular reason this parameter is not `const`?